graphite_api/functions.py

```diff
@@ -542,29 +568,43 @@ def asPercent(requestContext, seriesList, total=None):
 
     if total is None:
         totalValues = [safeSum(row) for row in zip_longest(*seriesList)]
-        totalText = None  # series.pathExpression
-    elif isinstance(total, list):
-        if len(total) != 1:
-            raise ValueError(
-                "asPercent second argument must reference exactly 1 series")
-        normalize([seriesList, total])
-        totalValues = total[0]
-        totalText = totalValues.name
+        totalText = "sumSeries(%s)" % formatPathExpressions(seriesList)
+    elif type(total) is list:
+        if len(total) != 1 and len(total) != len(seriesList):
+            raise ValueError("asPercent second argument must be missing, a "
+                             "single digit, reference exactly 1 series or "
+                             "reference the same number of series as the "
+                             "first argument")
+
+        if len(total) == 1:
+            normalize([seriesList, total])
+            totalValues = total[0]
+            totalText = totalValues.name
```

Comment:
Yes, as I understand it, the change was done to specifically support the case of `len(total) != 1` and `len(total) == len(seriesList)`